packages/ui/react-ui-form/src/hooks/useForm.ts

```diff
@@ -247,17 +248,24 @@ const flatMap = <T extends BaseObject>(errors: ValidationError[]) => {
   );
 };
 
-// TODO(ZaymonFC): Move to util?
-const getByPath = <T extends object, V>(obj: T, path: FormPath<T>): V => {
+const getByPath = <T extends object, V>(obj: Partial<T>, path: FormPath<T>): V | undefined => {
```

Comment:
i don't understand this notation: prop.3?
if this is for array indirection then use jsonpath notation 'prop[3]' and reuse the jsonpath lib.

Reply:
I don't understand this comment. Sorry. Can you clarify?

Reply:
how can the second part of the path be a number?